lib/internal/modules/cjs/loader.js

```diff
@@ -85,6 +85,20 @@ const relativeResolveCache = Object.create(null);
 
 let requireDepth = 0;
 let statCache = null;
+
+function enrichCJSError(err) {
+  const stack = err.stack.split('\n');
+
+  const lineWithErr = stack[1];
+
+  if (err.message.startsWith('Unexpected token export') ||
+    (/^\s*import(?=[ {'"*])\s*/).test(lineWithErr)) {
```

Comment:
Should just be `/^\s*import(?=[ {'"*])\s*(?![ (])/` actually. By including a space in the negative capture group we can avoid the backtracking.